libs/klv/src/main/java/org/codice/ddf/klv/KLV.java

```diff
@@ -0,0 +1,558 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.klv;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+/**
+ * A public domain class for working with Key-Length-Value (KLV)
+ * byte-packing and unpacking. Supports 1-, 2-, 4-byte, and BER-encoded
+ * length fields and 1-, 2-, 4-, and 16-byte key fields.
+ * <p>
+ * KLV has been used for years as a repeatable, no-guesswork technique
+ * for byte-packing data, that is, sending data in a binary format
+ * with two bytes for this integer, four bytes for that float,  and
+ * so forth. KLV is used in broadcast television and is defined in
+ * SMPTE 336M-2001, but it also greatly eases the burden of non-TV-related
+ * applications for an easy, interchangeable binary format.
+ * <p>
+ * The underlying byte array is always king. If you change the key
+ * length or the length encoding, you only change how the underlying
+ * byte array is interpreted on subsequent calls.
+ * <p>
+ * Everything in KLV is Big Endian.
+ * <p>
+ * All <tt>getValue...</tt> methods will return up to the number
+ * of bytes specified in the length fields unless there are fewer
+ * bytes actually given. In that case, the number of bytes given
+ * will be used. This is to make the code more robust for reading
+ * corrupted data.
+ *
+ * @author Robert Harder
+ * @author rharder # users.sourceforge.net
+ * @version 0.3
+ */
+public class KLV {
```

Comment:
For all of these classes that start with `KLV`, we camel case acronyms. So change them all to `Klv` instead.